examples/tutorial/jupyter/execution/pandas_on_dask/cluster/exercise_5.ipynb

```diff
@@ -0,0 +1,376 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from distributed import Client\n",
+    "\n",
+    "client = Client(cluster)\n",
+    "# Or use an IP address connection if the cluster instance is unavailable:\n",
+    "# client = Client(f\"{scheduler_adress}:8687\")\n",
+    "\n",
+    "client.wait_for_workers(n_workers)\n",
+    "client.ncores()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "After successful initialization of the cluster, you need to configure it.\n",
+    "\n",
+    "You can use plugins to install any requirements into workers:\n",
+    "* [InstallPlugin](\"https://distributed.dask.org/en/stable/plugins.html#distributed.diagnostics.plugin.InstallPlugin\")\n",
+    "* [PipInstall](\"https://distributed.dask.org/en/stable/plugins.html#distributed.diagnostics.plugin.PipInstall\")\n",
+    "* [CondaInstall](\"https://distributed.dask.org/en/stable/plugins.html#distributed.diagnostics.plugin.CondaInstall\").\n",
+    "\n",
+    "You have to install Modin package on each worker using ``PipInstall`` plugin."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import PipInstall\n",
+    "\n",
+    "client.register_plugin(PipInstall(packages=[\"modin\"]))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "If you need an additional workers configuration, you can create your own [WorkerPlugin](https://distributed.dask.org/en/stable/plugins.html#worker-plugins) or function that will be executed on each worker calling ``client.run()``.\n",
+    "\n",
+    "**NOTE**: Dask cluster does not check if this plugin or function has been called before. Therefore, you need to take this into account when using them.\n",
+    "\n",
+    "In this tutorial a CSV file will be read, so you need to download it to each of the workers and local machine in the same gloabal path. The following function has been implemented for this:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from dask.distributed import Worker\n",
+    "import os\n",
+    "import urllib\n",
+    "\n",
+    "def dataset_upload(file_url, file_path):\n",
+    "    try:\n",
+    "        dir_name = os.path.dirname(file_path)\n",
+    "        if not os.path.exists(dir_name):\n",
+    "            os.makedirs(dir_name)\n",
+    "        if os.path.exists(file_path): # os.path.isfile(file_path):\n",
+    "            return \"File has already existed.\"\n",
+    "        else:\n",
+    "            urllib.request.urlretrieve(file_url, file_path)\n",
+    "        return \"OK\"\n",
+    "    except Exception as ex:\n",
+    "        return str(ex)"
```

Comment:
```suggestion
    "    os.makedirs(filepath, exist_ok=True)
    "    urllib.request.urlretrieve(file_url, file_path)
```
Can we use something like this?

Reply:
We can, but in this case we will have some problems:
* if any exception is raised we won't receive any messages (Dask worker doesn't send any outputs so it must be managed by user)
* Since this function may be called more than once. we mast guarantee the correct and predictable result.

In my opinion my suggestion is better.